server/posts/tests/routes/comments.test.ts

```diff
@@ -0,0 +1,351 @@
+              color: null,
+              accessory:
+                "https://firebasestorage.googleapis.com/v0/b/bobaboard-fb.appspot.com/o/images%2Fbobaland%2Fundefined%2F9b7a5d90-4885-43bf-a5f5-e861b7b87505.png?alt=media&token=83ae88ca-5c81-4d1b-9208-0a936017c485",
+            },
+            user_identity: { name: "bobatan", avatar: "/bobatan.png" },
+            friend: false,
+            own: true,
+            new: true,
+          },
+        ],
+      };
+
+      expect(res.status).toBe(200);
+      expect(res.body).toEqual(expectedResponse);
+    });
+  });
+
+  test("allows a commenting as a reply to another comment", async () => {
+    await wrapWithTransaction(async () => {
+      setLoggedInUser(BOBATAN_USER_ID);
+      const commentId = "f2v1349d-da57-4703-8bab-54c12494e8b1";
+      jest.spyOn(uuid, "v4").mockReturnValueOnce(commentId);
+      const res = await request(server.app)
+        .post(`/${CHARACTER_TO_MAIM_POST_ID}/comments`)
+        .send(testReplyCommentBody);
+
+      const expectedResponse = {
+        comments: [
+          {
+            id: commentId,
+            parent_comment_id: KERMIT_FRIEND_COMMENT_ID,
+            chain_parent_id: null,
+            parent_post_id: CHARACTER_TO_MAIM_POST_ID,
+            created_at: expect.any(String),
+            content: '[{"insert":"society for maiming muppets when?"}]',
+            secret_identity: {
+              name: "Old Time-y Anon",
+              avatar:
+                "https://firebasestorage.googleapis.com/v0/b/bobaboard-fb.appspot.com/o/images%2Fgore%2F5c2c3867-2323-4209-8bd4-9dfcc88808f3%2Fd931f284-5c22-422d-9343-e509cfb44ffc.png?alt=media&token=94e52fff-4e6b-4110-94c3-90b8800f541c",
+              color: null,
+              accessory:
+                "https://firebasestorage.googleapis.com/v0/b/bobaboard-fb.appspot.com/o/images%2Fbobaland%2Fundefined%2F9b7a5d90-4885-43bf-a5f5-e861b7b87505.png?alt=media&token=83ae88ca-5c81-4d1b-9208-0a936017c485",
+            },
+            user_identity: { name: "bobatan", avatar: "/bobatan.png" },
+            friend: false,
+            own: true,
+            new: true,
+          },
+        ],
+      };
+
+      expect(res.status).toBe(200);
+      expect(res.body).toEqual(expectedResponse);
+    });
+  });
+
+  test("if the request's comment contents is not an array, throws a bad request error", async () => {
+    await wrapWithTransaction(async () => {
+      setLoggedInUser(BOBATAN_USER_ID);
+      const res = await request(server.app)
+        .post(`/${CHARACTER_TO_MAIM_POST_ID}/comments`)
+        .send(nonArrayContentsTestCommentBody);
+
+      expect(res.status).toBe(400);
+      expect(res.body).toEqual({
+        message: "Received non-array type as contents of comments.",
+      });
+    });
+  });
+
+  test("does not write to the db when the comment's content array is empty", async () => {
```

Comment:
I don't know if this is a useful test, but I stumbled upon this use case when I was working on testing the 'bad request' error. This confirms there's logic in place that means nothing happens to the db if the contents array is empty, but I don't know if that's beneficial to know. I'll remove if it seems weird/extra!

Reply:
I think it's beneficial to catch that we're changing the current behavior, so yes, thanks for this!

I would expect this to throw some kind of "your request was malformed" error, but I don't want you to make changes now so having the test at least documents the behavior.